coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/util/ThumbnailUtil.java

```
@@ -0,0 +1,183 @@
+        thumbnailCRS = "EPSG:3857";
+    }
+
+    public static String generateBase64Thumbnail(Item item) {
+        if(item.getItemType().equals(ItemType.uber)) {
+            log.error("Thumbnail generation is not supported for uber.");
+            return null;
+        }
+
+        try(ItemManager manager = new ItemManager()) {
+            BufferedImage baseMap = null;
+
+            // Transform and validate Bbox
+            Bbox transformedBox = Bbox.copyToCRS(Bbox.copyToSquareBox(item.getBbox()), thumbnailCRS);
+            String bboxVal = transformedBox != null ? transformedBox.getBbox() : null;
+
+            if(bboxVal == null || bboxVal.isEmpty()) {
+                log.error("Failed to retrieve Bbox for thumbnail for item: " + item.getId());
+                return null;
+            }
+
+            // Generate thumbanil
+            BufferedImage itemThumb = generateThumbnail(item, item.getId(), bboxVal, 1, manager, 0).getLeft();
+            
+            // If basemap WMS param configured try to pull base map image
+            if(thumbanilBaseWmsUrl != null && !thumbanilBaseWmsUrl.isEmpty()) {
+                baseMap = new BufferedImage(THUMBNAIL_SIZE, THUMBNAIL_SIZE, BufferedImage.TYPE_INT_RGB);
+                baseMap.getGraphics().drawImage(fetchBaseMap(bboxVal), 0, 0, THUMBNAIL_SIZE, THUMBNAIL_SIZE, null);
+            }
+
+            // Basemap fallback to white background
+            if(baseMap == null) {
+                baseMap = new BufferedImage(THUMBNAIL_SIZE, THUMBNAIL_SIZE, BufferedImage.TYPE_INT_RGB);
+                baseMap.getGraphics().fillRect(0, 0, THUMBNAIL_SIZE, THUMBNAIL_SIZE);
+            }
+            
+            // Draw generated thumb onto basemap
+            if(itemThumb != null) {
+                baseMap.getGraphics().drawImage(itemThumb, 0, 0, null);
+            } else {
+                log.error("Item " + item.getId() + " thumbnail failed to generate. Returning blank basemap.");
```

Did you mean to return something here?

---

No, at this point the only thing that has been drawn onto the image is the basemap so that's effectively what happens. The log is just to inform that instead of returning a blank image or an error for a failed thumbnail it'll return what has already been drawn on the image up to this point which is the basemap